src/core/src/energy/functions.rs

```diff
@@ -675,4 +772,33 @@ mod tests {
         let gaussian = Gaussian::new(8.0, -2.0);
         assert_eq!(gaussian.energy(0.0), -8.0);
     }
+
+    #[test]
+    fn test_mie() {
+        let mie = Mie::new(2.0, 0.8, 12.0, 6.0);
+        assert_eq!(mie.energy(2.0), 0.0);
+        assert_eq!(mie.energy(2.5), -0.6189584744448002);
+
+        assert_relative_eq!(mie.tail_energy(1.0), 1388.0888889, epsilon = 1e-6);
+        assert_relative_eq!(mie.tail_energy(2.0), -5.688888888888889, epsilon = 1e-6);
+        assert_relative_eq!(mie.tail_energy(14.42), -0.022767318648783084, epsilon = 1e-6);
+
+        assert_relative_eq!(mie.tail_virial(1.0), 17066.666666666668, epsilon = 1e-6);
+        assert_relative_eq!(mie.tail_virial(2.0), -17.06666666666667, epsilon = 1e-6);
+        assert_relative_eq!(mie.tail_virial(14.42), -0.1366035877536718, epsilon = 1e-6);
+
+        assert!(mie.force(f64::powf(2.0, 1.0 / 6.0) * 2.0).abs() < 1e-15);
+        assert_ulps_eq!(mie.force(2.5), -0.95773475733504);
+
+        let e0 = mie.energy(4.0);
+        let e1 = mie.energy(4.0 + EPS);
+        assert_relative_eq!((e0 - e1) / EPS, mie.force(4.0), epsilon = 1e-6);
+    }
+
+    #[test]
+    #[should_panic(expected = "The repulsive exponent n has to be larger than the attractive exponent m")]
+    fn test_mie_n_lower_m() {
+        let mie = Mie::new(2.0, 0.8, 6.0, 12.0);
+        assert_eq!(mie.energy(2.0), 0.0);
+    }
```

Comment:
Could you add a test for the tail correction with m < 3? Either checking for the panic or checking for a return value of 0